docs/proposal/aws-resource-handling.md

```diff
@@ -0,0 +1,200 @@
+
+## Existing solutions and drawbacks for our use
+
+### kops
+
+kops relies heavily on tagging of resources. This provides the benefit of recreating the state of the cluster resources by querying with filters. However, in many cases tagging requires a second API call and there are some resources that are not able to be tagged. If we succeed in creating a resource but fail to tag the resource, then we risk the chance of orphaning that resource for the user.
+
+### Kubicorn
+
+In contrast to the kops approach, Kubicorn mainly relies on recording the resource IDs as part of state. However, since we rely on an external API server for recording the state there is still a possibility of creating the resource and failing to record the resource ID, which still exposes the possible risk of orphaning that resource for the user.
+
+## Summary of edge cases for creating an individual resource
+
+1. resource create succeeds, but subsequent tagging fails
+2. resource creates succeeds, but update of cluster/machine object fails
+3. attempting to delete resource fails after an attempt to rollback due to a failure to record the ID of the created resource to the cluster/machine object for resources that do not support tagging on create.
+4. the controller/actuator dies after creating a resource but before tagging and or recording the resource
+
+## Misc TODOs
+
+- Incorporate retries for AWS or cluster-api requests?
+- Identify which resources fall into which class of workflow based on tagging support, client token support, etc.
+- Better define mutatable/non-mutatable attributes for objects during update
+
+## Using client tokens
+
+Where possible use [client tokens](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Run_Instance_Idempotency.html) in the create request so that subsequent requests will return the same response.
+
+## Tagging of resources
+
+Resources that are managed by the controllers/actuators should be tagged with: `kubernetes.io/cluster/<name or id>=owned`
+
+Question: Since other tools/components interact and create resources using the `kubernetes.io/cluster/<name or id>=owned` tag, should we also add a cluster-api-provider-aws specific tag for filtering?
+
+TODO: Define additional tags that can be used to provide additional metadata about the resource configuration/usage by the actuator. This is would allow us to rebuild status without relying on polluting the object config.
+
+## Handling of AWS api errors
+
+Each resource has specific error codes that it will return and these can be used to differentiate fatal errors from retryable errors. These errors are well documented in some cases (elbv2 api), and poorly in others (ec2 api). We should provide a best effort to [properly handle these errors](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/handling-errors.html) in the correct manner.
+
+## Proposed workflows
+
+### Resources that support tag on create support
+
+#### Create
```

Comment:
The Reconcile case is going to be slightly different since we might also be able to query by ID before calling create

Reply:
ahh i see ok 👍